### PR TITLE
Identify drives beyond the device file

### PR DESCRIPTION
**Pull request by niclan**
Identify drives beyond the device file

In this program we are using smartctl to get information anyway and it is a small step to also get the device brand/model/serial.

### CODE REVIEW

File: plugins/node.d/hddtemp_smartctl [hunk 1]
```diff
@@ -317,3 +322,39 @@ sub command_for_drive_device {
     $cmd .= $fulldev;
 
 }
+
+
+sub get_drive_id {
+    my ($drive, $fulldev, $use_nocheck) = @_;
+
+    print STDERR "ID of $drive\n";
```
Comment (sumpfralle) [inline plugins/node.d/hddtemp_smartctl]: This is probably a debug output that can be removed?

File: plugins/node.d/hddtemp_smartctl [hunk 2]
```diff
@@ -317,3 +322,39 @@ sub command_for_drive_device {
...
+	
+    my $id = '';
+    $id .= ' '. $family if $family;
+    $id .= ' '. "Model: $model" if $model;
+    $id .= ' '. "S/N: $serial" if $serial;
+    $id .= ' '. "WWN: $wwn" if $wwn;
+    $id = "Drive is a " . $id if $id;
+    
+    return $id;
+    exit 1;
```
Comment (sumpfralle) [inline plugins/node.d/hddtemp_smartctl]: This line is probably superfluous?

**Comment by niclan** [comment]
Sorry for mye lacking review. I belive the PR is clean now.

File: plugins/node.d/hddtemp_smartctl [hunk 3]
```diff
@@ -219,11 +219,14 @@ if (defined $ARGV[0]) {
     print "graph_category sensors\n";
     print "graph_info This graph shows the temperature in degrees Celsius of the hard drives in the machine.\n";
     foreach (@drives) {
+	my $d = clean_fieldname($_);
         my @dirs = splitdir($_);
-        print clean_fieldname($_) . ".label " . $dirs[-1] . "\n";
-        print clean_fieldname($_) . ".max 100\n";
-        print clean_fieldname($_) . ".warning 57\n";
-        print clean_fieldname($_) . ".critical 60\n";
+        print $d . ".label " . $dirs[-1] . "\n";
```
Comment (sumpfralle) [inline plugins/node.d/hddtemp_smartctl]: Maybe you could align the indentation here? (including the `splitdir` line above)

Comment (niclan) [reply plugins/node.d/hddtemp_smartctl]: I've spaced it (I'll need to reconstruct my .emacs file for official work..). I belive the shortest path forward now is that you select a squash merge - and make the commit comment something like "Identify drives beyond the device file" - at least we can do this in the GitHubEnterprise I'm used to. I'm not allowed to merge this PR to master myself.

Thanks

**Comment by sumpfralle** [comment]
I cherry-picked and squashed your commits as 1c8e35af95d109ab6657449d593eb748dbc418b5. Thank you!

**Comment by steveschnepp** [comment]
The "whitespace" issue isn't a tab/space one, it's a "trailing whitespace" one ;-)

**Pull request closed without merging**